operators/pkg/controller/elasticsearch/validation/validations.go

```diff
@@ -43,10 +43,13 @@ func nameLength(ctx Context) validation.Result {
 
 // supportedVersion checks if the version is supported.
 func supportedVersion(ctx Context) validation.Result {
-	if v := driver.SupportedVersions(ctx.Proposed.Version); v == nil {
-		return validation.Result{Allowed: false, Reason: unsupportedVersion(&ctx.Proposed.Version)}
+	if v := driver.SupportedVersions(ctx.Proposed.Version); v != nil {
+		if err := v.Supports(ctx.Proposed.Version); err == nil {
+			return validation.OK
+		}
 	}
-	return validation.OK
+	return validation.Result{Allowed: false, Reason: unsupportedVersion(&ctx.Proposed.Version)}
```

Comment:
Can you include a unit test in Test_supportedVersion() to cover what you changed? (which also makes sense outside the webhook world)